File: HomePage.java
package com.fl.automation.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import com.fl.automation.core.BrowserUtils;

public class HomePage {
    private WebDriver driver;

    private By findStoreLink = By.linkText("Find a Store");
    private By findStoreLinkFallback1 = By.cssSelector("header a[href*='stores.footlocker.com']");
    private By findStoreLinkFallback2 = By.xpath("//header//a[contains(.,'Find a Store') or contains(.,'Store Locator')]");

    public HomePage(WebDriver driver) {
        this.driver = driver;
    }

    public boolean isHomePageLoaded() {
        return driver.getTitle().contains("Foot Locker") || driver.getCurrentUrl().contains("footlocker.com");
    }

    public WebElement getFindStoreElement() {
        try {
            return driver.findElement(findStoreLink);
        } catch (Exception e1) {
            try {
                return driver.findElement(findStoreLinkFallback1);
            } catch (Exception e2) {
                return driver.findElement(findStoreLinkFallback2);
            }
        }
    }

    public void clickFindStore() {
        WebElement findStore = getFindStoreElement();
        BrowserUtils.click(driver, findStore);
    }
}

File: HomePage.py
# HomePage.py
"""
Page Object for the Foot Locker Homepage
Implements actions and validations for homepage elements
"""
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webdriver import WebDriver

class HomePage:
    def __init__(self, driver: WebDriver):
        self.driver = driver

    FIND_A_STORE_LINK = (By.XPATH, "//a[contains(text(), 'Find a Store')]")

    def click_find_a_store(self):
        find_a_store = self.driver.find_element(*self.FIND_A_STORE_LINK)
        find_a_store.click()

    def is_homepage_loaded(self):
        return self.driver.find_element(*self.FIND_A_STORE_LINK).is_displayed()
